### DISCUSSION TITLE
Multiple Servers / External Users

### DISCUSSION THREAD

**Discussion by Ramalama2** [dani-garcia/vaultwarden#4169] [Ideas]
Hi, i've searched like hell but can't find anything for this.

Is it possible or planned, to connect vaultwarden servers together somehow?

My personal issue is, that i have Vaultwarden running in my Company and Private.
Account switching is great on the Phone...
But useless on Desktops, since no Browser addon is supporting it and the "Desktop Client" doesnt support Passkeys.
The Desktop client is only usefull to open it and copy the content you need for the login, but a Passkey is nothing you can Copy...

Atm this isn't a big deal, since im the only one using Vaultwarden in the Company, no one is interrested at the moment and they still save their passwords directly in their Browsers, which is stupid to me.

Account Switching is a Feature, Bitwarden has since almost 1,5 Years, but all they've done is that useless Desktop Client and the Mobile Client.
What i want to say is, that i don't believe this will come to Browser Addons at all, or at least near future.

So a backend solution would be amazing.
Cheers :-)

**Comment by netboy3** [comment]
Bitwarden just released browser extension v2024.1.0 with the following new feature: "Log into multiple accounts and switch between them". I haven't tried it, but it might be the feature you are looking for.

**Comment by devNull2and1** [comment]
> Hi, i've searched like hell but can't find anything for this.
> 
> Is it possible or planned, to connect vaultwarden servers together somehow?
> 
> My personal issue is, that i have Vaultwarden running in my Company and Private. Account switching is great on the Phone... But useless on Desktops, since no Browser addon is supporting it and the "Desktop Client" doesnt support Passkeys. The Desktop client is only usefull to open it and copy the content you need for the login, but a Passkey is nothing you can Copy...
> 
> Atm this isn't a big deal, since im the only one using Vaultwarden in the Company, no one is interrested at the moment and they still save their passwords directly in their Browsers, which is stupid to me.
> 
> Account Switching is a Feature, Bitwarden has since almost 1,5 Years, but all they've done is that useless Desktop Client and the Mobile Client. What i want to say is, that i don't believe this will come to Browser Addons at all, or at least near future.
> 
> So a backend solution would be amazing. Cheers :-)

Hello, installed vaultwarden first time, love it, come from keepassxc with multiple databases open and synced with nextcloud between devices (clunky and error prone).

Would be awesome feature to connect other vaulwarden servers over tokens/secrets. I think it should be possible like this:
- add organization "myPrivateOrg" on vaultworden1.domain.com
- take token / secret from that organization
- add remote organization "myPrivateOrg" on vaultworden2.domain.com

**Comment by BlackDex** [comment]
That is probably not going to happen. Main reason is that there clients aren't build for this. And we would have to do significant modifications on the server side for this to work. Also, key rotations, new users all need to be synced and out of sync issues should be handled for.

I think this is too much out of scope for this project.

**Reply by Ramalama2** [reply]
If you have nothing agains, i would close this.

Because people are right, the client is getting support for multiple accounts. (I mean especially the Browser addons)
The only browser addon without multiuser support is safari atm... lol im so unlucky xD

But this will surely happen at some point either.

Otherwise vaultwarden is pretty amazing, most features that i wish would happen, is anyway all client related.
Cheers :-)

**Comment by devNull2and1** [comment]
> That is probably not going to happen. Main reason is that there clients sent build for this. And we would have to do significant modifications on the server side for this to work. Also, key rotations, new users all need to be synced and out of sync issues should be handled for.
> 
> I think this is too much out of scope for this project.

Hello @BlackDex thx for fast reply...it would be a great feature, the switching is not useful for many constellations because the browser plugin only finds matching entries from the active vaultwarden server and not across multiple servers

maybe there are other options?

**Reply by BlackDex** [reply]
Not really I think.
It would need to be configured via the backend, since there is no client frontend which can do this. That makes it difficult to also keep track of sync issues if they appear.

I'm at least not going to put time and effort in something like this  anytime soon (if ever).

**Comment by Ramalama2** [comment]
I think this can be closed, since almost all browser addons (except safari) support account switching.
Cheers
